ImportError: cannot import name 'Sas' from 'nio.crypto'

Originally reported: https://github.com/home-assistant/core/pull/101693#issuecomment-1753234552

On `0.22.0`: Importing `nio` without `python-olm` installed (aka installing `matrix-nio`, rather than `matrix-nio[e2e]` results in the following error:

```python
(venv-311) $ pytest tests/components/matrix/
ImportError while loading conftest '/.../tests/components/matrix/conftest.py'.
tests/components/matrix/conftest.py:8: in <module>
    from nio import (
../venv-311/lib/python3.11/site-packages/nio/__init__.py:9: in <module>
    from .client import *
../venv-311/lib/python3.11/site-packages/nio/client/__init__.py:3: in <module>
    from .base_client import Client, ClientConfig
../venv-311/lib/python3.11/site-packages/nio/client/base_client.py:35: in <module>
    from ..crypto import ENCRYPTION_ENABLED, DeviceStore, OlmDevice, OutgoingKeyRequest, Sas
E   ImportError: cannot import name 'Sas' from 'nio.crypto' (/.../venv-311/lib/python3.11/site-packages/nio/crypto/__init__.py)
```

--- a/nio/client/base_client.py
+++ b/nio/client/base_client.py
@@ -19,6 +19,7 @@ from collections import defaultdict
 from dataclasses import dataclass, field
 from functools import wraps
 from typing import (
+    TYPE_CHECKING,
     Any,
     Awaitable,
     Callable,
@@ -32,7 +33,7 @@ from typing import (
     Union,
 )
 
-from ..crypto import ENCRYPTION_ENABLED, DeviceStore, OlmDevice, OutgoingKeyRequest, Sas
+from ..crypto import ENCRYPTION_ENABLED, DeviceStore, OutgoingKeyRequest
 from ..events import (
     AccountDataEvent,
     BadEvent,
@@ -76,14 +77,11 @@ from ..rooms import MatrixInvitedRoom, MatrixRoom
 if ENCRYPTION_ENABLED:
     from ..crypto import Olm
     from ..store import DefaultStore, MatrixStore, SqliteMemoryStore
+if TYPE_CHECKING:
+    from ..crypto import OlmDevice, Sas
+
 
 from ..event_builders import ToDeviceMessage
-
-try:
-    from json.decoder import JSONDecodeError
-except ImportError:
-    JSONDecodeError = ValueError  # type: ignore
-
 
 logger = logging.getLogger(__name__)
 

